Merge from toCode to master

tgargula opened the pull request.

Commit:
Add pdf files

Commit:
Update README.md

Commit:
Add simple graph algorithms

Commit:
Add topological_sorting.py and euler_cycle.py

jakub-gonet commented on ASD/graph/DFS.py:
```diff
@@ -0,0 +1,56 @@
+    u.t1 = time
+    u.color = "grey"
+    for v in G.neighbors(u):
+        if v.color == "white":
```
Why do you need those colors? `is_visited` field should suffice

tgargula replied:
I've written this file as DFS template so I decided not to replace 'colors' with 'visited'

jakub-gonet commented on ASD/graph/topological_sorting.py:
```diff
@@ -0,0 +1,48 @@
+    for v in G.neighbors(u):
+        if v.color == "grey":   # CYCLE
+            print("Graph is not a DAG")
+            sys.exit()
```
you should raise an exception here

Commit:
add Millers 15 task - jebany kwestionariusz gdzie o chuj chodzi

mrrys00 committed:
modify gitignore

mrrys00 committed:
add new homework from Miller - not finished

mrrys00 committed:
something with Millers PI

Commit:
add ASD10 tasks

Commit:
Update compile and memcheck

– add warning flags and debug flag in 'compile'
– update 'memcheck' so it works with .c and .cpp arguments

Commit:
Update branch

Commit:
Add graph algorithms in .cpp

Commit:
Add pdf file with tasks

Commit:
add tasks 09 form Miller

Commit:
Merge branch 'toCode' of https://github.com/tgargula/AGH into toCode

Commit:
Add exception

Commit:
Update local

Wolokin reviewed (approved):
Cosmetics ;)

Commit:
11th tasks off/ online

Commit:
add Miller's 10th tasks

tgargula commented on PI/Miller/10/linkedListFromTxt.c:
```diff
@@ -0,0 +1,147 @@
+    
+    word[i2] = '\0';
+    if (isLetter(word[0]) == 0) return; 
+    char *s = (char*)malloc(strlen(word) * sizeof(char));
```
It is unnecessary if you use char s[MAXCHAR].
But if you want to allocate it dynamically:
`char \*s = (char\*)malloc(strlen(word) + 1) * sizeof(char));`
This allocated memory is not freed!

tgargula commented on PI/Miller/10/linkedListFromTxt.c:
```diff
@@ -0,0 +1,147 @@
+    }
+    
+    if(listIterator != NULL && strcmp(word, listIterator->word) == 0) {
+        listIterator->counter += 1; return first;
```
Same as above

tgargula commented on PI/Miller/10/linkedListFromTxt.c:
```diff
@@ -0,0 +1,147 @@
+    printf("Number of word occurences: %d\n", first->counter);
+}
+
+int main() {
```
It is better to write:
`int main(void) {`

Commit:
Add lab08 tasks

tgargula commented on ASD/graph/BFS.py:
```diff
@@ -0,0 +1,54 @@
+                v.d = u.d + 1
+                v.parent = u
+                Q.put(v)
+        u.color = "black"
```
But it actually is.
At first we put `s` in the empty queue. Then process all of its neighbours and at last we change its color to black.

tgargula commented on ASD/graph/BFS.py:
```diff
@@ -0,0 +1,54 @@
+        self.value = value
+        self.parent = None
+        self.d = inf
+        self.color = "white"
```
I agree with that, but I have written this code as a BFS template

Commit:
Update BFS.py and DFS.py

Commit:
Update README.md

Commit:
improve codes

Commit:
Merge branch 'toCode' of https://github.com/tgargula/AGH into toCode

tgargula commented on PI/Miller/10/linkedListFromTxt.c:
```diff
@@ -95,10 +99,11 @@ node* processFile(node* first) {                        // przetwarzanie pliku
     }
 
     fclose(txtfile);
+    free(inputFileName);
```
Invalid free – inputFileName was not dynamically allocated

Commit:
improve codes
